src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
 public class FileWork {
+
+    private static final String DELIMITERS = "[^A-Za-z0-9]";
+    private static final String SPECIFIED_CHARACTER = "w";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            int value = bufferedReader.read();
+            if (value == -1) {
+                return new String[]{};
+            }
+            while (value != -1) {
+                builder.append((char) value);
+                value = bufferedReader.read();
+            }
```

Comment:
you're appending to `builder` just to split it afterwards at line 27. let's process data after it is read.